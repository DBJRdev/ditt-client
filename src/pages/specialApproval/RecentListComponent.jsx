import { Link } from 'react-router-dom';
import React from 'react';
import { withTranslation } from 'react-i18next';
import Layout from '../../components/Layout';
import SpecialApprovalList from '../../components/SpecialApprovalList';
import routes from '../../routes';
import styles from './specialApproval.scss';

const ListComponent = (props) => (
  <Layout title={props.t('specialApproval:title.recentSpecialApprovals')}>
    <div className={styles.infoText}>
      {props.t('specialApproval:text.recentSpecialApprovalsBeginningPart')}
      &nbsp;
      {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
      <Link to={routes.specialApprovalList}>
        {props.t('specialApproval:text.recentSpecialApprovals')}
      </Link>
      &nbsp;
      {props.t('specialApproval:text.recentSpecialApprovalsClosingPart')}
    </div>
    <SpecialApprovalList {...props} />
  </Layout>
);

ListComponent.defaultProps = SpecialApprovalList.defaultProps;
ListComponent.propTypes = SpecialApprovalList.propTypes;

export default withTranslation()(ListComponent);
