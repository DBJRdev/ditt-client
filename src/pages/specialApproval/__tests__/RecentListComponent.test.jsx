import Immutable from 'immutable';
import React from 'react';
import { shallow } from 'enzyme';
import RecentListComponent from '../RecentListComponent';
import configMock from '../../../../tests/mocks/configMock';

describe('rendering', () => {
  it('renders correctly', () => {
    const props = {
      config: configMock,
      fetchBusinessTripWorkLog: () => {},
      fetchConfig: () => {},
      fetchHomeOfficeWorkLog: () => {},
      fetchOvertimeWorkLog: () => {},
      fetchSpecialApprovalList: () => {},
      fetchSpecialLeaveWorkLog: () => {},
      fetchTimeOffWorkLog: () => {},
      fetchVacationWorkLog: () => {},
      isFetching: false,
      isPosting: false,
      markBusinessTripWorkLogApproved: () => {},
      markBusinessTripWorkLogRejected: () => {},
      markHomeOfficeWorkLogApproved: () => {},
      markHomeOfficeWorkLogRejected: () => {},
      markMultipleSpecialLeaveWorkLogApproved: () => {},
      markMultipleSpecialLeaveWorkLogRejected: () => {},
      markMultipleVacationWorkLogApproved: () => {},
      markMultipleVacationWorkLogRejected: () => {},
      markOvertimeWorkLogApproved: () => {},
      markOvertimeWorkLogRejected: () => {},
      markSpecialLeaveWorkLogApproved: () => {},
      markSpecialLeaveWorkLogRejected: () => {},
      markTimeOffWorkLogApproved: () => {},
      markTimeOffWorkLogRejected: () => {},
      markVacationWorkLogApproved: () => {},
      markVacationWorkLogRejected: () => {},
      specialApprovalList: Immutable.fromJS({
        businessTripWorkLogs: [],
        homeOfficeWorkLogs: [],
        overtimeWorkLogs: [],
        specialLeaveWorkLogs: [],
        timeOffWorkLogs: [],
        vacationWorkLogs: [],
      }),
      token: 'token',
    };
    const tree = shallow(<RecentListComponent {...props} />);

    expect(tree).toMatchSnapshot();
  });
});
