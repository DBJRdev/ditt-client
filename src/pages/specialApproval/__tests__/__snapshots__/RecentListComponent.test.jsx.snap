// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<Connect(LayoutComponent)
  title="specialApproval:title.recentSpecialApprovals"
>
  <div
    className="infoText"
  >
    specialApproval:text.recentSpecialApprovalsBeginningPart
     
    <Link
      to="/special-approvals"
    >
      specialApproval:text.recentSpecialApprovals
    </Link>
     
    specialApproval:text.recentSpecialApprovalsClosingPart
  </div>
  <SpecialApprovalListComponent
    businessTripWorkLog={null}
    config={
      Immutable.Map {
        "supportedHolidays": Immutable.List [
          "2018-01-01T00:00:00.000Z",
          "2018-12-24T00:00:00.000Z",
          "2018-12-25T00:00:00.000Z",
          "2018-12-26T00:00:00.000Z",
          "2018-12-31T00:00:00.000Z",
        ],
        "supportedYears": Immutable.List [
          2018,
          2019,
          2020,
          2021,
        ],
        "workedHoursLimits": Immutable.Map {
          "lowerLimit": Immutable.Map {
            "changeBy": -1800,
            "limit": 21600,
          },
          "upperLimit": Immutable.Map {
            "changeBy": -2700,
            "limit": 32400,
          },
        },
      }
    }
    fetchBusinessTripWorkLog={[Function]}
    fetchConfig={[Function]}
    fetchHomeOfficeWorkLog={[Function]}
    fetchOvertimeWorkLog={[Function]}
    fetchSpecialApprovalList={[Function]}
    fetchTimeOffWorkLog={[Function]}
    fetchVacationWorkLog={[Function]}
    homeOfficeWorkLog={null}
    isFetching={false}
    isPosting={false}
    markBusinessTripWorkLogApproved={[Function]}
    markBusinessTripWorkLogRejected={[Function]}
    markHomeOfficeWorkLogApproved={[Function]}
    markHomeOfficeWorkLogRejected={[Function]}
    markMultipleVacationWorkLogApproved={[Function]}
    markMultipleVacationWorkLogRejected={[Function]}
    markOvertimeWorkLogApproved={[Function]}
    markOvertimeWorkLogRejected={[Function]}
    markTimeOffWorkLogApproved={[Function]}
    markTimeOffWorkLogRejected={[Function]}
    markVacationWorkLogApproved={[Function]}
    markVacationWorkLogRejected={[Function]}
    overtimeWorkLog={null}
    specialApprovalList={
      Immutable.Map {
        "businessTripWorkLogs": Immutable.List [],
        "homeOfficeWorkLogs": Immutable.List [],
        "overtimeWorkLogs": Immutable.List [],
        "timeOffWorkLogs": Immutable.List [],
        "vacationWorkLogs": Immutable.List [],
      }
    }
    t={[Function]}
    timeOffWorkLog={null}
    token="token"
    vacationWorkLog={null}
  />
</Connect(LayoutComponent)>
`;
